File: lib/m3u8_player/site_kit.rb
module M3u8Player
  module SiteKit
    BASE_URL = "https://m3u8-player.net"

    module_function

    def home_url
      BASE_URL
    end

    def tool_url(slug)
      clean = slug.to_s.gsub(%r{\A/+|/+\z}, "")
      clean.empty? ? BASE_URL : "#{BASE_URL}/#{clean}/"
    end

    def player_url
      "#{BASE_URL}/#player"
    end

    def downloader_url
      tool_url("m3u8downloader")
    end

    def converter_url
      tool_url("m3u8tomp4")
    end

    def hls_player_url
      tool_url("hls-player")
    end
  end
end

File: nuget/M3u8PlayerSiteKit/SiteKit.cs
namespace M3u8PlayerSiteKit;

public static class SiteKit
{
    public const string BaseUrl = "https://m3u8-player.net";

    public static string HomeUrl() => BaseUrl;

    public static string ToolUrl(string slug)
    {
        var clean = slug.Trim('/');
        return clean.Length == 0 ? BaseUrl : $"{BaseUrl}/{clean}/";
    }

    public static string PlayerUrl() => $"{BaseUrl}/#player";

    public static string DownloaderUrl() => ToolUrl("m3u8downloader");

    public static string ConverterUrl() => ToolUrl("m3u8tomp4");

    public static string HlsPlayerUrl() => ToolUrl("hls-player");

    public static string IptvPlayerUrl() => ToolUrl("iptv-player");
}
